### DISCUSSION TITLE
Jakarta REST 3.2 plan in RESTEasy

### DISCUSSION THREAD

**Discussion by anbusampath** [resteasy/resteasy#4036] [General]
Based on rest-dev mailing [thread](https://www.eclipse.org/lists/rest-dev/msg00206.html) mostly Jakarta EE 11 going to include REST 3.2 while reading RESTEasy [roadmap](https://resteasy.dev/2023/12/06/overview-and-roadmap/) noticed REST 4 going to be released. Is there plan change to support REST 3.2 in RESTEasy?

**Comment by jamezp** [comment]
@anbusampath Yes. We are actively working with the specification and RESTEasy will be Jakarta REST 3.2 compatible. It's not completely clear yet if Jakarta EE 11 will include Jakarta REST 3.2 or stick with Jakarta REST 3.1.

My personal preference will be to use Jakarta REST 3.2 so that we can formally deprecate the `@Context` and `@Suspended` annotations instead of simply removing them.

My plan was to update the roadmap once the decision was made for what is going into Jakarta EE 11. Maybe I should add something to it now though to help with any confusion and be more transparent in what is happening.

**Reply by anbusampath** [reply]
Thanks. Hopefully Jakarta EE 11 will include Jakarta REST 3.2🤞.

**Reply by jamezp** [reply]
Thank you for the input. I've add a note to the blog post https://resteasy.dev/2023/12/06/overview-and-roadmap/.

**Reply by jamezp** [reply]
Just an FYI the new play will be a Jakarta REST 4.0, however a much more stripped down version of what it was going to be. See https://github.com/jakartaee/rest/issues/1233
